source/vsignal/slot.d

```diff
@@ -0,0 +1,156 @@
+module vsignal.slot;
+
+/**
+Binds a function to a Slot. A payload can be passed if the function to call is
+a data member function of the same or if the function accepts as its first
+parameter a reference to its type.
+
+Params:
+	pred = the function to bind.
+	slot = the Slot that holds the function.
+	instance = the payload to keep.
+*/
+package void connect(alias pred, F)(auto ref Slot!F slot)
+{
+	import vsignal.utils : tryForward;
+
+	with(slot)
+	{
+		payload = null;
+		fn = (const scope void*, Parameters args) {
+			static if (is(ReturnType == void))
+				pred(tryForward!(pred, args));
+			else
+				return ReturnType(pred(tryForward!(pred, args)));
+		};
+	}
+}
+
+@safe pure nothrow @nogc unittest
+{
+	Slot!(void delegate(int)) slot;
+
+	class C
+	{
+		static void opCall(int) {}
+		static void fooA(const int) {}
+		static void fooB(ref int) {}
+		static void fooC(const ref int) {}
+	}
+
+	// all lambda listeners can omit the type
+	slot.connect!((ref _)       {});
+	slot.connect!((const ref _) {});
+	slot.connect!((_)           {});
+	slot.connect!((const _)     {});
+
+	slot.connect!(C.opCall);
+	slot.connect!(C.fooA);
+	slot.connect!(C.fooB);
+	slot.connect!(C.fooC);
+}
```

Comment:
Does this unittest test something?

Reply:
It shows you can connect all those listeners with those signatures without failing. Because this is an internal thing there is not much to test. The actual connection tests are done at the higher-level structures.